src/components/pages/lesson.js

```diff
@@ -238,14 +247,19 @@ function Lesson({ PageTemplate, initialState }) {
   const publicStorage = {
     rootPath: 'media',
     initialPath: `media/${lesson._id}`,
-    uploadPath: `media/${lesson._id}`
+    uploadPath: `media/${lesson._id}`,
+    isDeletionEnabled: hasUserPermission(user, permissions.DELETE_ANY_STORAGE_FILE)
   };
 
-  const privateStorage = isPrivateRoom && !!user.storage.plan
+  const privateStorage = isPrivateRoom && !!room.owner.storage?.plan
     ? {
+      usedBytes: room.owner.storage.usedBytes,
+      maxBytes: room.owner.storagePlan.maxBytes,
       rootPath: `rooms/${room._id}/media`,
       initialPath: `rooms/${room._id}/media`,
-      uploadPath: `rooms/${room._id}/media`
+      uploadPath: `rooms/${room._id}/media`,
+      isDeletionEnabled: isRoomOwner || isRoomCollaborator,
+      onUsedBytesUpdated: handleUsedBytesUpdated
```

Comment:
I opted for putting this callback here to be less intrusive with this patch fix 😬

Reply:
Good idea to not pollute all components in between!

Reply:
Was not original :D